### PR TITLE
[JIT] Dont use specialized tensor type

### PR DESCRIPTION
**Pull request by eellison**
Fix for https://github.com/pytorch/pytorch/issues/46122

For `Any`, we infer the type of the ivalue to set the ivalue's type tag. When we saw a Tensor, we would use a specialized Tensor type, so when `Dict[str, Tensor]` was passed in as any `Any` arg it would be inferred as `Dict[str, Float(2, 2, 2, 2)]` which breaks runtime `isinstance` checking.

### CODE REVIEW

File: test/test_jit.py [hunk 1]
```diff
@@ -7396,6 +7396,18 @@ def any_refinement2(a):
         self.assertEqual(any_refinement2(3), torch.tensor(3))
         self.assertEqual(any_refinement2(torch.tensor(5)), torch.tensor(5))
 
+    def test_unspecialized_any_binding(self):
+        # any binding will infer the type, if it infers
+        # a specialized tensor type `x` Dict type will fail isinstance check
+
+        @torch.jit.script
+        def foo(x: Any):
+            assert isinstance(x, Dict[str, torch.Tensor])
```
Comment (gmagogsfm) [inline test/test_jit.py:7405]: Is there a way at all to express a more specified Tensor type in Python? If so, does this PR break checking against that?

Comment (eellison) [reply test/test_jit.py:7405]: No, there's not

File: torch/csrc/jit/python/pybind_utils.h [hunk 2]
```diff
@@ -256,8 +256,7 @@ InferredType tryToInferContainerType(py::handle input);
 inline InferredType tryToInferType(py::handle input) {
   // Try tensor types
   if (THPVariable_Check(input.ptr())) {
-    auto tensor = py::cast<at::Tensor>(input);
-    return InferredType(TensorType::create(tensor));
+    return InferredType(TensorType::get());
```
Comment (gmagogsfm) [inline torch/csrc/jit/python/pybind_utils.h:259]: TensorType::getInferred()?

Comment (eellison) [reply torch/csrc/jit/python/pybind_utils.h:259]: TensorType::getInferred is a typing-level construct, which says that an unannotated variable is being inferred to be a Tensor. IMO this doesn't have a lot to do with using a `Tensor` for an `Any` arg

**Review by gmagogsfm — APPROVED** [review]

**Pull request closed without merging**